packages/react-dropdown/src/Items.js

```diff
@@ -100,8 +102,12 @@ export default function DropdownItems({
 }: Props) {
   return (
     <Items>
-      {items.map((item, index) => {
-        const itemIndex = item.hasOwnProperty('index') ? item.index : index;
+      {items.map((item: DropdownItemWithIndex, index: number) => {
+        let itemIndex = index;
+        if (item.hasOwnProperty('index')) {
+          itemIndex = item.index;
+          delete item.index;
```

Comment:
I think the way you defined a new item object previously was easier to understand.

Reply:
I have to remove the index at some point. Can you propose what you mean exactly containing the delete part?

Reply:
do you mean this: 
```
const itemIndex = item.hasOwnProperty('index') ? item.index : index;
delete item.index;
```

Reply:
I see that previously you created a new object to remove the index:

```
item: {
  id: item.id,
  label: item.label,
},
```

can't you do the same?

Reply:
I need to pass the same stuff that item has expect index. For example if user provides items like these:
```
const items = {
   id: 1,
   label: 'Hey',
   puppy: 'huge'
}
```

We have to make sure to pass it back the same way.

What I could do is:

```
item: {
  ...item
  id: item.id,
  label: item.label,
},
```
but I should remove `index`. I think. What do you think?

Reply:
in that case I'd go with a spread operator:

```
const { index, ...itemWithoutIndex } = item;
```